src/modules/ekf2/EKF2.cpp

```diff
@@ -1376,62 +1340,9 @@ void EKF2::UpdateRangeSample(ekf2_timestamps_s &ekf2_timestamps)
 	}
 }
 
-void EKF2::UpdateMagCalibration(const hrt_abstime &timestamp)
+void EKF2::SaveMagDeclination()
 {
-	// Check if conditions are OK for learning of magnetometer bias values
-	// the EKF is operating in the correct mode and there are no filter faults
-	if (_ekf.control_status_flags().in_air && _ekf.control_status_flags().mag_3D && (_ekf.fault_status().value == 0)) {
-
-		if (_last_magcal_us == 0) {
-			_last_magcal_us = timestamp;
-
-		} else {
-			_total_cal_time_us += timestamp - _last_magcal_us;
-			_last_magcal_us = timestamp;
-		}
-
-		// Start checking mag bias estimates when we have accumulated sufficient calibration time
-		if (_total_cal_time_us > 30_s) {
```

Comment:
We have effectively removed this 30 second requirement. This is probably OK given the checks on variances, but will need to be checked with a series of short up and down flights including some yaw rotations.

Reply:
It's now handled by a hysteresis. https://github.com/PX4/PX4-Autopilot/pull/16417/files#diff-82b382c1f89f118ca86cd0ad5765be02769ce61703f41079537eee1635aa34c5R111

Reply:
Nevermind, I kept it in ekf2 to properly handle 30 seconds of accumulated 3D mag fusion time.